include/aws/cryptosdk/cache.h

```diff
@@ -531,14 +531,42 @@ AWS_CRYPTOSDK_STATIC_INLINE void aws_cryptosdk_materials_cache_release(
  * @param cache_limit_ttl_units The units of cache_limit_ttl. Allowable values are defined in aws/common/clock.h
  */
 AWS_CRYPTOSDK_API
-struct aws_cryptosdk_cmm *aws_cryptosdk_caching_cmm_new(
+struct aws_cryptosdk_cmm *aws_cryptosdk_caching_cmm_new_from_cmm(
     struct aws_allocator *alloc,
     struct aws_cryptosdk_materials_cache *materials_cache,
     struct aws_cryptosdk_cmm *upstream,
     const struct aws_byte_buf *partition_id,
     uint64_t cache_limit_ttl,
     enum aws_timestamp_unit cache_limit_ttl_units);
 
+/**
+ * Creates a new instance of the caching crypto materials manager. This CMM will intercept requests for encrypt
+ * and decrypt materials, and forward them to the provided materials cache.
+ *
+ * If multiple caching CMMs are attached to the same materials_cache, they will share entries if and only if the
+ * partition_id parameter is set to the same string. Unless you need to use this feature, we recommend passing NULL, in
```

Comment:
minor: This line seem to be overflowing github viewing limit, could you run the style-formatting script to limit the number of characters to 120?

Reply:
I agree that these particular lines are a bit long, but I literally copied the exact same lines from the other caching CMM constructor documentation and just modified the description of the one parameter that is different.

I'm fine with modifying the clang-format settings to change the allowed line lengths, but that will potentially affect a lot more files throughout the code base, so it should be done in a separate code review.

Note that all of these document comments will be rendered in the doxygen web pages. I need to put a PR up for that separately because it goes into another branch of the repo. But when I put that PR up after this one, you will be able to read the new wording added here more easily on the webpages, so you'll have a second chance to review the wording here if you want to change something.

I'll manually change the lengths of these lines this one time to make them easier to read, but in the future you can also just pull my branch into your local git repo to read it there rather than on github's browser interface.

Reply:
I didnot notice that the prior lines also had this issue, we should probably make the change to those lines as well to be consistent.